internal/p2p/peermanager.go

```diff
@@ -1254,6 +1324,40 @@ func (p *peerInfo) Copy() peerInfo {
 	return c
 }
 
+// LastDialed returns when the peer was last dialed, and if that dial
+// attempt was successful.
+func (p *peerInfo) LastDialed() (time.Time, bool) {
+	var (
+		last    time.Time
+		success bool
+	)
+	last = last.Add(-1) // so it's after the epoch
```

Comment:
I don't think we need both the flag and this adjustment trick: If we have the flag, we can use zero as the baseline (since that's what it will wind up being anyway); and if we have a non-epoch baseline we can use that to detect when we've made an update, e.g.,

```go
return last, !last.Add(1).IsZero()
```

or words to that effect. (I kinda prefer the latter, but I don't think we need both)

Reply:
in the case when the most recent dialing attempt was a failure, last will be non-zero but the second value will need to be false.